homeassistant/components/greeneye_monitor/__init__.py

```diff
@@ -117,34 +120,62 @@
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the GreenEye Monitor component."""
-    monitors = greeneye.Monitors()
-    hass.data[DATA_GREENEYE_MONITOR] = monitors
-
-    server_config = config[DOMAIN]
-    await monitors.start_server(server_config[CONF_PORT])
-
-    async def close_monitors(event: Event) -> None:
-        """Close the Monitors object."""
-        await monitors.close()
+    if server_config := config.get(DOMAIN):
```

Comment:
Seems like both YAML and UI configs are supported. I think the best practice is to migrate to UI if you find a YAML config.
Searching the code base for `async_step_import` would give you some examples.